assets/js/base/hooks/use-store-cart-item-quantity.js

```diff
@@ -45,7 +47,10 @@ export const useStoreCartItemQuantity = ( cartItem ) => {
 	// Observe debounced quantity value, fire action to update server when it
 	// changes.
 	useEffect( () => {
-		changeCartItemQuantity( cartItem.key, debouncedQuantity );
+		// Don't run it if quantity didn't change but it was set for the first time.
+		if ( Number.isFinite( previousDebouncedQuantity ) ) {
```

Comment:
I do have some concern here about `previousDebouncedQuantity` not being included as a dependency (so there is risk of it going stale). But in this case, I think the staleness is _desirable_ because that's what is actually fixing the issue.

Reply:
Shouldn’t we compare previous with current value? or am I miss understanding usePrevious

Reply:
I think we're taking advantage of the stale props issue here where at the time the effect is first called, `previousDebouncedQuantity` is `undefined`. Hence it's suitable as a method for preventing the initial quantity set from triggering a request.  Subsequent calls to useEffect won't be affected at all by the conditional because `previousDebouncedQuantity` will always be a number.

Reply:
would it make sense to add an extra to check to not trigger debounced values that are the same, in shippingRates I simply compared debouncedValues with the previous values and bailed if they’re the same.

Reply:
[React docs](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect) seem to suggest all values used inside the effect should be declared in the dependencies, but I'm not sure that applies to our case because we don't want to run the effect when `previousDebouncedQuantity` changes, only when `debouncedQuantity` changes. Otherwise, the fix wouldn't work.

Reply:
> would it make sense to add an extra to check to not trigger debounced values that are the same

I don't think that's needed. `QuantitySelector` doesn't trigger `onChange` if the introduced value is the same that it was, and using the `-` and `+` buttons will always change the value.